### PR TITLE
add simulation script for withdrawing fund from treasury

### PR DESCRIPTION
**Pull request by RoseRompkxm**
## Description

A simple simulation of withdrawing fund from treasury, making sure everything is good

Resolves #

## Checklist

- [ ] I have updated the documentation to account for the changes in the code.
- [ ] If I added new functionality, I added tests covering it.
- [ ] If I fixed a bug, I added a test preventing this bug from silently reappearing again.
- [ ] My contribution follows [Venus contribution guidelines](docs/CONTRIBUTING.md).

### CODE REVIEW

File: script/hardhat/utils/misc.js [hunk 1]
```diff
@@ -11,19 +11,24 @@ function deploy(contract, ...arguments) {
...
 async function impersonate(addr) {
     await network.provider.request({
         method: "hardhat_impersonateAccount",
         params: [addr],
     });
-    await network.provider.send("hardhat_setBalance", [
-        addr,
-        "0x10000000000000000000000",
-    ]);
+    await setBalance(addr, IMPERSONATION_STARTING_BALANCE);
```
Comment (kkirka) [inline script/hardhat/utils/misc.js:25]: 👍

**Comment by Thenewguy911** [comment]
Yikes

**Review by kkirka — APPROVED** [review]

**Comment by kkirka** [comment]
> Yikes

It's a simulation that ensures that Timelock (governance) can withdraw treasury funds to a multisig wallet. It's needed for the grants program outlined in this proposal:
https://snapshot.org/#/venus-xvs.eth/proposal/0x15decfa3258764c93d5cfb9ec1d378673892a59476d27ed7456a1973cc69c69c

**Comment by Thenewguy911** [comment]
Do you think its important to create new snapshots at the present time and allocate the limited resources Venus has  to that before you honor your prior commitments such as vai issue and vrt swap?

**Comment by kkirka** [comment]
> Do you think its important to create new snapshots at the present time and allocate the limited resources Venus has to that before you honor your prior commitments such as vai issue, new tokenomics implemetation, shortfall, and vrt swap?

Of course it is. We have a pipeline of features, and the smart contract development for the items you mentioned is already completed. We are releasing these gradually and preparing for the next items in the queue.

**Comment by Thenewguy911** [comment]
You havent released one of them yet.  So it's hard to understand why work is being done on brand new items with multiple items spoke of not released.  For example there was 7 days between last vrt githib updates for very minor changes thst were already audited, that we both understand could have taken a day.  The code here is not super complicated but its been worked on for months on end at a snales pace.  I'm sure there is some financial benefit to the team delaying these items for months, and that's very disappointing.

**Pull request merged**